### PR TITLE
Add mapbox to snakemake workflow

### LINKED ISSUE #115: Move web mercator transform for mapbox to project_orthos.py
This is basically duplicate code. Let's do all of the transformation at once. We should probably add a level of directory structure in `projected_orthos` with one directory for each projection type. Alternatively we can keep the current structure and include the name of the projection in the file name. `upload_mapbox.py` can then load the files from the appropriate location.

### PR DESCRIPTION
**Pull request by ethanwhite**
* change upload_mapbox.py to run on one file at a time
* move webmercator projection to project_orthos.py
* use mapbox API to check if file is already uploaded and require a force flag to reupload
* add mapbox steps to Snakefile

Closes #115 #113 #98 #96

### CODE REVIEW

**Comment by ethanwhite** [comment]
I've tested the outcome from this on our Shiny apps and everything seems to work well. Once this is in we should cleanup the .mbfiles that are in the root of `/blue/ewhite/everglades/mapbox/` since the new location follows the `year/site/` structure of the other directories.

File: Zooniverse/upload_mapbox.py [hunk 1]
```diff
@@ -1,70 +1,55 @@
...
+    with open("/blue/ewhite/everglades/mapbox/mapbox.ini", "rb") as f:
+        toml_dict = tomli.load(f)
+        token = toml_dict['mapbox']['access-token']
+    api_base_url = "https://api.mapbox.com/v4"
+    tileset_id = f"bweinstein.{flight}"
+    url = f"{api_base_url}/{tileset_id}.json?access_token={token}"
+    response = requests.get(url)
+    if response.status_code == 200:
+        return True
+    else:
```
Comment (henrykironde) [inline Zooniverse/upload_mapbox.py:21]: we can do without the `else` statement

File: Zooniverse/upload_mapbox.py [hunk 2]
```diff
@@ -1,70 +1,55 @@
...
-                src.crs, dst_crs, src.width, src.height, *src.bounds)
-            kwargs = src.meta.copy()
-            kwargs.update({
-                    'crs': dst_crs,
-                    'transform': transform,
-                    'width': width,
-                    'height': height
-            })
+def on_mapbox(flight):
+    """Check if the mbtiles file is has already been uploaded to mapbox"""
```
Comment (henrykironde) [inline Zooniverse/upload_mapbox.py:11]: is has

**Review by henrykironde — COMMENTED** [review]
Looks good to me, just a few things we could improve on

**Comment by ethanwhite** [comment]
Good catches @henrykironde. I took the return simplification one step further following https://adamj.eu/tech/2020/01/17/simplify-your-ifs-that-return-booleans/. I used a new commit instead of cleanup and rebasing because I'm already branched off of this and don't want to go through all of the rebase conflicts again.

**Pull request merged**